build-support/bin/download_binary.sh

```diff
@@ -25,16 +30,43 @@ readonly util_name="$2"
 readonly version="$3"
 readonly filename="${4:-${util_name}}"
 
-os=$("${REPO_ROOT}/build-support/bin/get_os.sh")
-readonly path="bin/${util_name}/${os}/${version}/${filename}"
+function make_download_path_for_os {
+  local os_string="$1"
+  echo "bin/${util_name}/${os_string}/${version}/${filename}"
+}
+
+function curl_output_file_with_fail {
+  local download_url="$1"
+  local output_path="$2"
+  echo >&2 "Downloading ${download_url} ..."
+  curl --fail "$download_url" > "$output_path"
+}
+
+path="$(make_download_path_for_os "$("${REPO_ROOT}/build-support/bin/get_os.sh")")"
 readonly cache_path="${CACHE_ROOT}/${path}"
 
+maybe_alternate_os="$([[ "$(uname)" == 'Darwin' ]] \
```

Comment:
This function only adds compatibility for macOS, right? It wouldn't help with Windows or some Linux distro? If so, consider renaming to something like `maybe_macos` to be more explicit.

Reply:
This is absolutely a correct comment. In the commit I just pushed, we use BinaryUtil to avoid having to duplicate this regardless, but this is absolutely correct.